core/sql/optimizer/ControlDB.cpp

```diff
@@ -108,7 +108,12 @@ ControlDB::~ControlDB()
 
 void ControlDB::setRequiredShape(ControlQueryShape *shape)
 {
-  delete requiredShape_;
+  if (requiredShape_)
+    {
+      delete requiredShape_->getShape();
+      delete requiredShape_;
```

Comment:
@selvaganesang can you explain this more? When I look at the definition of NADELETE, it explicitly says that NADELETE is **not** meant for NABasicObjects, and we are dealing with NABasicObjects here. Is it that you don't trust the method in which we put the heap pointer into NABasicObjects?